app/javascript/controllers/dashboard_date_range_controller.js

```diff
@@ -53,7 +53,7 @@ export default class extends Controller {
               .add(-1, 'month')
               .endOf('month')
           ],
-          'This Month': [moment().startOf('month'), moment()],
+          'This Month': [moment().startOf('month'), moment().endOf('month')],
```

Comment:
Not sure this will address the end date stored in session.

Reply:
It should - happy to elaborate.